### PR TITLE
Fix docblocks and better IDE and static analyzer discoverery.

### PR DESCRIPTION
**Pull request by dereuromark**
I was surprised to see a public trait message not being part of the interface.
This should get fixed with 4.0 then.

### CODE REVIEW

File: src/Collection/CollectionTrait.php [hunk 1]
```diff
@@ -885,9 +892,10 @@ public function _unwrap()
     }
 
     /**
-     * {@inheritDoc}
```
Comment (ADmad) [inline src/Collection/CollectionTrait.php]: The `{@inheritDoc}` needs to be kept for inheriting the description.

File: src/Collection/Iterator/FilterIterator.php [hunk 2]
```diff
@@ -43,7 +43,7 @@ class FilterIterator extends Collection
      * in the current iteration, the key of the element and the passed $items iterator
      * as arguments, in that order.
      *
-     * @param \Iterator $items The items to be filtered.
+     * @param \Iterator|\Traversable|array $items The items to be filtered.
```
Comment (ADmad) [inline src/Collection/Iterator/FilterIterator.php]: `Iterator` extends `Traversable`, so have both is redundant. `\Iterator` can be removed now.

**Review by ADmad — CHANGES_REQUESTED** [review]

**Comment by ADmad** [comment]
> I was surprised to see a public trait message not being part of the interface.
> This should get fixed with 4.0 then.

`CollectionInterface::cartesianProduct()` has already been added on 4.x.

File: src/Collection/CollectionTrait.php (hunk 1 above)
Comment (dereuromark) [reply src/Collection/CollectionTrait.php]: there is nothing to inherit. i didnt find any code.

**Pull request merged**